docs/okeps/okep-5217-routed-ingress-l2-udn-without-ipam.md

```diff
@@ -0,0 +1,116 @@
+# OKEP-5217: Routed ingress support for L2 UDN without dynamic IP allocation
+
+* Issue: [#5217](https://github.com/ovn-org/ovn-kubernetes/issues/5217)
+
+## Problem Statement
+
+There is an increased interest for VMs in Kubernetes, which is generating the
+desire to import VMs from another virtualization platforms.
+
+VMs running in virtualization platforms are usually using static IP allocation,
+where IPs are assigned directly in the guest.
+
+One of the features most sought after by virtualization users is routed ingress
+for L2 UDN, something which - right now - OVN-Kubernetes allows using the BGP
+feature.
+
+The current gap in OVN-Kubernetes is to provide routed ingress to VMs whose IPs
+were assigned statically in the guest. This enhancement plans to bridge that
+gap.
+
+## Goals
+
+- Detect/expose the MAC for the VM to be configured in OVNK for the VM (or disable port security)
+- When creating the UDN, be able to specify the gateway IP address, so the migrated VM can keep the same default route.
+- When creating the UDN, allow excludeSubnets to be used with L2 or L3 UDNs to ensure OVNK does not use an IP address from the range that VMs have already been assigned outside of the cluster.
```

Comment:
Isn't this enhancement just for layer2 ? I was not expected layer3 to be mention here, in fact this is specified at the limitations.

Also if we are going to discover the VM's internal IP then we don't need the excludeSubnets at UDN, ovn-k should just reserve it after reading for whatever mechanism we decide here, or are we talking about other IPs ?